cmake/RobotologySuperbuildOptions.cmake

```
@@ -3,6 +3,8 @@
 
 set(YCM_USE_CMAKE_PROPOSED TRUE CACHE BOOL "Use files including unmerged cmake patches")
 
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Let's define the variable as an option, otherwise users can't override this setting if they like (I used the same style used in https://github.com/Kitware/CMake/blob/d6f32880c6b36e614a887b641be8103560c1e9e8/Modules/CMakeGenericSystem.cmake#L75C1-L78C50):

```suggestion
  set(CMAKE_EXPORT_COMPILE_COMMANDS "ON"
    CACHE BOOL "Enable/Disable output of compile commands during generation."
    )
  mark_as_advanced(CMAKE_EXPORT_COMPILE_COMMANDS)
```